misc/azure-release.yml

```diff
@@ -6,7 +6,7 @@ stages:
       LIBTILEDB_VERSION: dev
       LIBTILEDB_SHA: dev
     ${{ else }}: 
-      TILEDBPY_VERSION: 0.10.4
+      TILEDBPY_VERSION: 0.10.5
       LIBTILEDB_VERSION: 2.4.3
       LIBTILEDB_SHA: 5cb082458f6e4214d5ae5beeeb667ec2231991ab
```

Comment:
Need to update the SHA too

Reply:
wait nvm

Reply:
SHA should be the same since it's still on 2.4.3.